morpho-checkpoint-node/src/Morpho/Ledger/SnapshotTimeTravel.hs

```diff
@@ -9,32 +10,39 @@ module Morpho.Ledger.SnapshotTimeTravel
 where
 
 import Cardano.Prelude hiding (Handle, ReadMode, atomically, to, withFile)
+import qualified Data.FingerTree.Strict as FT
+import Morpho.Ledger.Block
 import Ouroboros.Consensus.Ledger.Basics
 import Ouroboros.Consensus.Ledger.Extended
 import Ouroboros.Consensus.Storage.ChainDB.API
 import Ouroboros.Consensus.Util.IOLike
-import Ouroboros.Network.AnchoredFragment (anchorPoint)
+import Ouroboros.Network.AnchoredFragment (unanchorFragment)
 import Ouroboros.Network.Block hiding (Tip)
+import qualified Ouroboros.Network.ChainFragment as CF
 
-newtype TimeTravelError blk = LedgerStateNotFoundAt (Point blk)
+data TimeTravelError blk = LedgerStateNotFoundAt (Point blk)
+                         | ChainNotLongEnough Int Int
   deriving (Show)
 
 getLatestStableLedgerState ::
-  (MonadIO m, MonadSTM m) =>
+  (MonadIO m, MonadSTM m, HasHeader (Header blk)) =>
   ChainDB m blk ->
+  Int ->
   m (Either (TimeTravelError blk) (LedgerState blk))
-getLatestStableLedgerState chainDB = go (5 :: Int)
+getLatestStableLedgerState chainDB offset = do
+    chainFragment <- unanchorFragment <$> atomically (getCurrentChain chainDB)
+    let result = CF.lookupByIndexFromEnd chainFragment offset
+    case result of
+          FT.Position _ h _ -> getLedger h
+          _                 -> return $ Left $ ChainNotLongEnough offset (CF.length chainFragment)
+    -- There is a race condition here. If the ledger db advances a lot between the two
```

Comment:
This comment is supposed to be linked to `getPastLedger` line 45, we should move it line 44.